docs/DafnyRef/Topics.md

```diff
@@ -1146,3 +1146,24 @@ method TestD(dd: D) {
+
+There are particular expressions which, for technical reasons, Dafny can not use as part of a trigger. Among others, these expression include: match, let, arithmetic operations and logical connectives. For example, in the quantifier `forall x :: x in S ⇐⇒ f(x) > f(x+1)`, Dafny will use `x in S` and `f(x)` as trigger terms, but will not use `x+1` or any terms that contain it. You can investigate which triggers Dafny can not use by hovering over quantifiers in the IDE and looking for 'Rejected triggers', or by using the options `--show-tooltips` when using the LCI.
+
+Besides not finding triggers, another problematic situation is when Dafny was able to generate triggers, but believes the triggers it found may still cause useless instantiations because they create matching loops. Dafny emits a warning when this happens, indicating the Dafny program should be changed so Dafny can find triggers for this quantifier that do not cause matching loops.
+
+To understand matching loops, one needs to understand how triggers are used. During verification, Dafny maintains a set of expression which it believes to be true, which we call ground terms. Given a univeral quantifier that's a ground term, Dafny will use its triggers to pattern match on other ground terms. If the pattern matches, that ground term is used to instantiate the quantifier. 
+
+Dafny makes sure not to perform the exact same instantiation twice. However, if an instantiation leads to a new term that also matches the trigger, but is different from the term used for the instantiation, the quantifier may be instantiated too often, which we call a matching loop. For example, given the ground terms `f(3)` and `forall {f(x)} x :: f(x) + f(f(x))`, where `{f(x)}` indicates the trigger for the quantifier, Dafny may instantiate the quantifier using `3` for `x`. This creates a new ground term `f(3) + f(f(3))`, of which the right hand side again matches the trigger, allowing Dafny to instantiate the quantifier again using `f(3)` for `x`, leading to an unbounded amount of instantiations.
+
+Even existential quantifiers need triggers. This is because when Dafny determines an existential quantifier is false, for example in `requires !exists x :: f(x) == 2`, Dafny will use a logical rewrite rule to change this existential into a universal quantifier, so it becomes `requires forall x :: f(x) != 2`. Before verification, Dafny can not determine whether quantifiers will be determined to be true or false, so it must assume any quantifier may turn into a universal quantifier, and thus they all need triggers.
+
+Besides quantifiers, comprehensions such as set and map comprehension also need triggers, since these are modelled using universal quantifiers.
```

Comment:
```suggestion
Besides quantifiers, comprehensions such as set and map comprehension also need triggers, since these are modeled using universal quantifiers.
```
US English is used throughout the documentation (see Color)